src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/PyrisConnectorService.java

```diff
@@ -97,15 +119,123 @@ public void executePipeline(String feature, String variant, Object executionDTO)
      * @param variant      The variant of the feature to execute
      * @param executionDTO The DTO sent as a body for the execution
      */
-    public void executeLectureWebhook(String variant, PyrisWebhookLectureIngestionExecutionDTO executionDTO) {
+    public void executeLectureAddtionWebhook(String variant, PyrisWebhookLectureIngestionExecutionDTO executionDTO) {
         var endpoint = "/api/v1/webhooks/lectures/" + variant;
         try {
             restTemplate.postForEntity(pyrisUrl + endpoint, objectMapper.valueToTree(executionDTO), Void.class);
         }
+        catch (HttpStatusCodeException e) {
+            log.error("Failed to send lecture unit {} to Pyris: {}", executionDTO.pyrisLectureUnit().lectureUnitId(), e.getMessage());
+            throw toIrisException(e);
+        }
+        catch (RestClientException | IllegalArgumentException e) {
+            log.error("Failed to send lecture unit {} to Pyris: {}", executionDTO.pyrisLectureUnit().lectureUnitId(), e.getMessage());
+            throw new PyrisConnectorException("Could not fetch response from Pyris");
+        }
+    }
+
+    /**
+     * uses getLectureUnitIngestionState for all lecture units and then determines the IngestionState of the lecture
+     *
+     * @param courseId id of the course
+     * @return The ingestion state of the lecture
+     *
+     */
+    public Map<Long, IngestionState> getLecturesIngestionState(long courseId) {
+        Set<Lecture> lectures = lectureRepository.findAllByCourseId(courseId);
+        return lectures.stream().collect(Collectors.toMap(DomainObject::getId, lecture -> getLectureIngestionState(courseId, lecture.getId())));
+
+    }
+
+    /**
+     * uses getLectureUnitIngestionState for all lecture units and then determines the IngestionState of the lecture
+     *
+     * @param courseId  id of the course
+     * @param lectureId id of the lecture
+     * @return The ingestion state of the lecture
+     *
+     */
+    private IngestionState getLectureIngestionState(long courseId, long lectureId) {
+        Map<Long, IngestionState> states = getLectureUnitsIngestionState(courseId, lectureId);
+
+        if (states.values().stream().allMatch(state -> state == IngestionState.DONE)) {
+            return IngestionState.DONE;
+        }
+
+        if (states.values().stream().allMatch(state -> state == IngestionState.NOT_STARTED)) {
+            return IngestionState.NOT_STARTED;
+        }
+
+        if (states.values().stream().allMatch(state -> state == IngestionState.ERROR)) {
```

Comment:
Just making sure, is this intended?
```suggestion
        if (states.values().stream().anyMatch(state -> state == IngestionState.ERROR)) {
```

Reply:
yes it is intended, because that is the state of the ingestion of a lecture that has multiple lecture units